notebooks/image_models/solutions/conditional_gan.ipynb

```diff
@@ -0,0 +1,1139 @@
+{
```

Comment:
<p>It feels still too much code at one. Here is a way to simplify:</p><p> </p><ul><li class="ql-indent-1">Define the following method as standalone functions outside the class. This will give you the opportunity to explain in markdown cells what they do and will make the overall code intake much more palatable</li><li class="ql-indent-2"> <code>generator_train_step(generator, batch_size, input_digits)</code>  </li><li class="ql-indent-2"> 	<code>discriminator_train_step(discriminator, batch_size, input_digits)</code></li><li class="ql-indent-2"> 	<code>rain_step(self, features)</code></li></ul><p> </p><ul><li class="ql-indent-1">Consider switching to a custom training loop directly instead of packaging the custom training code into a Keras model. Is there any advantage of packaging the custom training code into a Keras Model that? It feels to me that it adds complexity and remove opportunities to explain what is really happening in the training loop.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/222/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Reply:
<p>Thanks for your input!</p><p>I moved this subclassing model to the optional section, and added a custom training loop functions in the main section.</p><p>Please take a look at that part again.</p>